### PR TITLE
Add KNN and MIP functions

### PR DESCRIPTION
**Pull request by hangsuUNC**
Hi Kiran,

I rebased the branch and added a few functions about MIP and KNN stuff in assemble.rs. Could you please help to merge this branch to main? 

Thanks,

Hang

### CODE REVIEW

File: src/hidive/Cargo.toml [hunk 1]
```diff
@@ -34,4 +34,41 @@ serde = "1.0.204"
 serde_json = "1.0.120"
 skydive = { path = "../skydive" }
 url = "2.5.2"
-#xgboost = "0.1.4"
\ No newline at end of file
+#xgboost = "0.1.4"
+ndarray = "0.15"
+linfa = "0.5"
+<<<<<<< HEAD
```
Comment (kvg) [inline src/hidive/Cargo.toml:40]: Looks like there are some unresolved rebase issues here causing the build to fail. Can you address these?

File: src/skydive/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,10 @@
 [package]
 name = "skydive"
+<<<<<<< HEAD
 version = "0.1.58"
+=======
```
Comment (kvg) [inline src/skydive/Cargo.toml:5]: Here too.

**Review by kvg — CHANGES_REQUESTED** [review]

**Comment by hangsuUNC** [comment]
Hi, I resolved some changes and hopefully there will be no conflicts now...

Please let me know if you need anything else!

Thanks,

Hang

**Comment by kvg** [comment]
Thanks! Don't forget to mark your branch as ready for review!

**Comment by kvg** [comment]
There seems to be a build failure related to this commit - is there something else we need to add to the Docker image before this will build successfully?

https://github.com/broadinstitute/hidive/actions/runs/10324946547/job/28585508145?pr=20#step:5:6739

**Comment by hangsuUNC** [comment]
> There seems to be a build failure related to this commit - is there something else we need to add to the Docker image before this will build successfully?
> 
> https://github.com/broadinstitute/hidive/actions/runs/10324946547/job/28585508145?pr=20#step:5:6739

It seems the build failed to install rust-wfa2 with `#12 210.9 error: failed to run custom build command for wfa2-sys v0.1.0 (https://github.com/pairwise-alignment/rust-wfa2.git#4cbfeb3d)`. I can build and compile this package from github repository in the local laptop though...

**Comment by kvg** [comment]
I see.  Can you work with @SHuang-Broad or @bshifaw to get this build issue resolved before we merge this in?

**Pull request merged**